lib/python/qmk/cli/new/keyboard.py

```diff
@@ -0,0 +1,105 @@
+"""This script automates the generation of a new keyboard directory.
+"""
+import os
+import shutil
+
+from distutils.dir_util import copy_tree
+from datetime import datetime
+from milc import cli
+
+
+def replace_files(filenames, search, replace):
+    """Replaces keywords across multiple files
+    """
+    for filename in filenames:
+        with open(filename, 'r+') as file:
+            contents = file.read()
+            file.seek(0)
+            file.write(contents.replace(search, replace))
+            file.truncate()
+
+@cli.argument('-kb', '--keyboard', help='Specify keyboard name. Example: 1upkeyboards/1up60hse')
+@cli.argument('-t', '--type', help='Specify keyboard type. [avr|ps2avrgb]')
+@cli.argument('-u', '--username', help='Specify username. Example: skullydazed')
+@cli.subcommand('Creates a new keyboard of your choosing')
+def new_keyboard(cli):
+    """Creates a new keyboard of your choosing.
+    """
+    # ask for user input if keyboard, type, or username was not provided in the command line
+    keyboard = cli.config.new_keyboard.keyboard if cli.config.new_keyboard.keyboard else input("Keyboard Name: ")
+    keyboard_type = cli.config.new_keyboard.type if cli.config.new_keyboard.type else input("Keyboard Type: ")
+    username = cli.config.new_keyboard.username if cli.config.new_keyboard.username else input("Username: ")
+
+    # check inputs
+    valid = True
+    if not keyboard:
+        cli.log.error('Keyboard is a required argument')
+        valid = False
+    if not username:
+        cli.log.error('Username is a required argument')
+        valid = False
+    valid_keyboard_types = ["avr", "ps2avrgb"]
+    if keyboard_type not in valid_keyboard_types:
+        cli.log.error('Keyboard type %s must be one of %s!', keyboard_type, valid_keyboard_types)
+        valid = False
+    
+    if not valid:
+        exit(1)
```

Comment:
Another nit-picking, sry. :)
Here you set a variable and fail later. Between line 55 and 63 is a similar structure, but there you fail immediately.
I have no preference for one or the other, but probably would be best to stick with one, at least in the same file.

Reply:
Made it the same. 👍